### ISSUE TITLE
Add support for Zanata API

### ISSUE THREAD

**Issue by brcisna** [BingLingGroup/autosub#54]
Hello All,

Just a  thought. Would it be possible for you to script as an option into autosub to have Zanata translate server as a possible translation source? Not having a paid Google Translate account(API key)  to be able to translate from English to Spanish on the google translate sever makes things kind of rough for casual users.
In other words,,still use google to generate the subtitle,,then hookup to zanata server APi (Free) to do the translation into any language. Just a thought. 

Thank You

**Comment by BingLingGroup** [comment]
You should read the readme carefully. I said it is free to translate by google now because autosub is using py-googletrans to translate the subtitles. Of cource it's possible to add your api to autosub.

**Comment by brcisna** [comment]
Hello,

Thanks for replying, 
when i try and translate from english to spanish with 
autosub file.mp4 -S en -D es
The process starts and reaches the translaitning stage, then fails saying " for translations an Google Translate API Key is required. and fails of course.
Also on my systen Debian ,i had to install the 'origin' version as the 'alpha branch' install would end up throwing an Openssl.sslsystemcallerror(32.'EPIPE')
Using the origin version autosub everything processes correctly., but throws the 'translations requires an Google Translate API key. if trying a different -D output source.

Hope this makes sense.
Thank You

**Comment by BingLingGroup** [comment]
@brcisna Please give me a full log about the sslsystemcallerror. Otherwise it is difficult to locate the error.
Just copy all the message from the terminal is OK.
And about the origin branch's function, I have already said in the readme that all the new features only available in alpha branch. And the origin branch currently is not maintained. So it is impossible for you to translate in alpha branch version unless you offer the API key. About the Google Translation V2 API, I never try to test it out. If you want to use it, you need to do it yourself.

**Comment by brcisna** [comment]
Hello BingLingGroup,

After doing some checking online, in regards to the alpha branch build of autosub throwing the OpenSSL.ssl.systemcall (32.'EPIPE') error,,this is fairly well documented that it does do a traceback,,but is caused by a network timeout. This would happen at the end phase of doing the translation. This was being caused by the no Google Translation API Key being present. It was just not failing with a good error message. The origin build v-0.4.1 after trying to translate would in fact end with an " To translate you are required to have a Google Translate API Key". which makes sense.
Also it appears that the py-googletrans script stopped working about eight months ago.

Thanks again

**Comment by BingLingGroup** [comment]
@brcisna Are you sure that py-googletrans can't work now? I tried it and it worked very well just two weeks ago. Otherwise I won't add this support.

I add py-googletrans after 0.5.0a. So it is useless for you to prove that py-googletrans is not working by testing on 0.4.1a or the version before.

I just need the **full log** of the 0.5.1a to locate problem. No log no bug.

Just copy all the message from the terminal is OK.

**Comment by BingLingGroup** [comment]
I just test the 0.5.1a. Everything works fine on my computer.

```
autosub -i "C:\userfile\Video\autosub\test\temp\[Crappy 1080P] Code Lyoko S4E66.ass" -SRC en -D zh-tw -hsp -bm all
No output format specified. Use input format "ass" for output.
Input is a subtitles file.

Translating text from "en" to "zh-tw".
Translation: 100% |########################################################################################################################| Time:  0:00:36 
Destination language subtitles file created at "C:\userfile\Video\autosub\test\temp\[Crappy 1080P] Code Lyoko S4E66.zh-tw.ass".

All works done.
```

You can check this [solution](https://github.com/ssut/py-googletrans/issues/123) about changing the request url though the default url works fine on my computer. You can use `-surl` to change the default request url in autosub.

**Comment by brcisna** [comment]
Hello BingLingGroup,

Thank you for helping out with this. Yes,it certainly appears the translations are working on your machine. Windows. 
I am using Debian Buster here at my work place. At home I am using Debian Stretch , just for some complete information. The two different version do fail slightly different in the post translation message? I try both alpha branch & origin branch 

WIll attach two screen shots. One is doing a translate from an .mp4 video directly. The second SS is after transcribing the .srt file to English,,(saved) then sending this .srt file to be translated.to Spanish / es  (Two steps) ,,just to see if this would work. 
Curious,do you have an Google Translate API key( service paid for)? Probably not as it doesn't appear you have the -K parameter, in your command line.

OK, Looking i guess there is not an option to attach photos/screenshots ,here?
IN the end both tries ends in a 443 request, "Cannot connect to Proxy"connection refused  and the first translation from an actual (video.mp4) ends with simply 'translation failed".. 

Thank You

**Comment by brcisna** [comment]
Terminnal output:

```
autosub -i imleaving.en.srt -SRC en -D es -hsp -bm all
No output format specified. Use input format "srt" for output.
Input is a subtitles file.

Translating text from "en" to "es".
Translation: N/A% |                                             | ETA:  --:--:--Traceback (most recent call last):
  File "/usr/local/bin/autosub", line 10, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/dist-packages/autosub/__init__.py", line 150, in main
    styles_list=None)
  File "/usr/local/lib/python2.7/dist-packages/autosub/cmdline_utils.py", line 631, in subs_trans
    service_urls=args.service_urls
  File "/usr/local/lib/python2.7/dist-packages/autosub/core.py", line 325, in list_to_googletrans
    src=src_language)
  File "/usr/local/lib/python2.7/dist-packages/googletrans/client.py", line 172, in translate
    data = self._translate(text, dest, src)
  File "/usr/local/lib/python2.7/dist-packages/googletrans/client.py", line 75, in _translate
    token = self.token_acquirer.do(text)
  File "/usr/local/lib/python2.7/dist-packages/googletrans/gtoken.py", line 200, in do
    self._update()
  File "/usr/local/lib/python2.7/dist-packages/googletrans/gtoken.py", line 57, in _update
    r = self.session.get(self.host)
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 546, in get
    return self.request('GET', url, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 646, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/adapters.py", line 510, in send
    raise ProxyError(e, request=request)
requests.exceptions.ProxyError: HTTPSConnectionPool(host='translate.google.com', port=443): Max retries exceeded with url: / (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f5ad89d6410>: Failed to establish a new connection: [Errno 111] Connection refused',)))
````

**Comment by BingLingGroup** [comment]
@brcisna There's a huge difference between the origin branch and the alpha branch. You can check it in my [readme](https://github.com/BingLingGroup/autosub#branches).

After reading the log, I guess I know the problem. Perhaps your proxy has some problems to connect to. `-hsp` means set the https proxy to `https://127.0.0.1:1080`. You need to ensure there's actually a local https server that is running on the port 1080 and it can accept https request.

If you can directly access to Google, you don't need to set the proxy server.

If you want to attach any screenshots, just dragging files at the bottom of the reply box.

**Comment by brcisna** [comment]
HI BLG

Here is terminal output without the parameter -hsp. (I only added the -hsp switch to replicate your commandline to see if it would work for me.)

 autosub -i imleaving.en.srt -SRC en -D es
No output format specified. Use input format "srt" for output.
Input is a subtitles file.

Translating text from "en" to "es".
Translation: 100% |#############################################| Time:  0:00:00
Error: Translation failed.


 This is the alpha branch 0.51a autosub 


Thank You

**Comment by BingLingGroup** [comment]
@brcisna 

![20190813083503](https://user-images.githubusercontent.com/42505588/62907330-48ca9400-bda5-11e9-84f3-373355de1818.png)

![20190813083619](https://user-images.githubusercontent.com/42505588/62907369-76174200-bda5-11e9-96bd-5d4b9f7b3448.png)

I test the case of `en` to `es` translation. It works well.

I hope you can test the `py-googletrans` externally to see if there's anything wrong on `py-googletrans`.

Install `py-googletrans` by using pip.
```
pip install googletrans
```

Open Python interpreter. Following is the input.

```python
python
>>> from googletrans import Translator
>>> translator = Translator()
>>> result = translator.translate('This is a test', src='en', dest='es')
>>> print(result.text)
```

Following is the output.
```python
Esto es una prueba
```

If it fails, it means `py-googletrans` may encounter some problem working on your computer. If not, it means autosub 0.5.1a takes the blame.

**Comment by earthcoin123** [comment]
@BingLingGroup
Translate srt file is fail,Translate ass file is success.

**Comment by BingLingGroup** [comment]
@brcisna It seems like a bug I fixed in #62. You can try the latest version to see if yours has been solved.

**Comment by BingLingGroup** [comment]
It seems Zanata is just a platform for translators to translate documents. I hope to use a free translation api instead of a platform or community. https://github.com/zanata/zanata-platform
